slapo/primitives/checkpoint.py

```diff
@@ -0,0 +1,55 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Checkpoint primitive."""
+# pylint: disable=arguments-differ
+
+from torch import nn
+
+from .base import register_primitive, Primitive
+from ..checkpoint import checkpoint as checkpoint_module
+
+
+@register_primitive()
+class CheckpointPrimitive(Primitive):
+    """Add activation checkpointing to the entire module or a subgraph.
+
+    Parameters
+    ----------
+    subgraph : Optional[List[List[torch.fx.Node]]]
+        The subgraph to be checkpointed. If None, checkpoint the entire module.
+    order_args_fn : Optional[Callable]
+        A function to order the position and keyword arguments of the module.
+        If None, the arguments will be ordered by the order of the position arguments,
+        followed by the key order of the keyword arguments.
+    """
+
+    @staticmethod
+    def name():
+        return "checkpoint"
```

Comment:
Can we use `@property` instead of a `@staticmethod`?

Reply:
Maybe. I'll try.

Reply:
Found that it's not that straightforward without adding other dependencies because we want to make it static. I'll keep the current implementation for now.